app/hotels/src/allHotels/searchForm/locationPicker/Types.js

```diff
@@ -0,0 +1,6 @@
+// @flow
```

Comment:
I really don't like doing this. It's confusing. Every component defines its own properties and I don't see a reason why should they be shared (they look the same but there is no relation between these components, right?). It's fine to repeat yourself here. But maybe you have a different opinion on this...

Reply:
> they look the same but there is no relation between these components
That is not how I see it. 

`LocationPickerScreen` loads locations and passes them to `RecentSearches` who passes a location to `RecentSearchItem`,  this are the same locations. If this object changes in the future, say we add a `searchedDate` we would have to change the type in 3 places.

Maybe a separate file is not the way to do it, maybe export it directly from `LocationPickerScreen`?

Reply:
I understand that but my point is this: component `RecentSearches` defines its properties and `RecentSearchItem` defines its properties. From the component perspective, there is no relation between these two components because they are encapsulated and if something change in the parent component it cannot affect the inner component - you don't have to change everything, just change one level and keep to contract with the second level. We did this mistake with `AvailableHotelSearchInput` type and it's the biggest mess ever.

Anyway, I think it's better to export the type from the other component.